From: Antoine Musso <hashar@free.fr>
Date: Mon, 6 May 2019 21:38:30 +0200
Subject: WMF: backport Don't call the merger for non-live items

Backport of upstream https://review.opendev.org/#/c/643703/

Stripped from:
* accumulating files to the item, there is no such thing in Zuul v2.5
* tests since the test suite is broken in our fork :/

Bug: T140297
Change-Id: Iaba6f3a0e99a264e0c1882e13beea6eedfbf8006
---
 zuul/scheduler.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zuul/scheduler.py b/zuul/scheduler.py
index bf2f12e..435e347 100644
--- a/zuul/scheduler.py
+++ b/zuul/scheduler.py
@@ -1596,7 +1596,7 @@ class BasePipelineManager(object):
                 change_queue.moveItem(item, nnfi)
                 changed = True
                 self.cancelJobs(item)
-            if actionable:
+            if actionable and item.live:
                 ready = self.prepareRef(item)
                 if item.current_build_set.unable_to_merge:
                     failing_reasons.append("it has a merge conflict")
